pkg/cmd/waitforkube/waitforkube.go

```diff
@@ -0,0 +1,57 @@
+package waitforkube
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+const (
+	retryDuration = 10 * time.Second
+	saTokenPath   = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+)
+
+type waitForKubeOpts struct {
+	errOut io.Writer
+}
+
+// NewWaitForKubeCommand waits for kube to come up before continuing to start the rest of the containers.
+func NewWaitForKubeCommand(errOut io.Writer) *cobra.Command {
+	waitForKubeOpts := &waitForKubeOpts{
+		errOut: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:   "wait-for-kube",
+		Short: "wait for kube service account to exist",
+		Long:  "This command makes sure that the kube is available before starting the rest of the containers in the pod.",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+						fmt.Fprint(waitForKubeOpts.errOut, err.Error())
+					}
+				}
+			}
+			must(waitForKubeOpts.Run)
+		},
+	}
+
+	return cmd
+}
+
+func (w *waitForKubeOpts) Run() error {
+	wait.PollInfinite(retryDuration, func() (bool, error) {
+		if _, err := os.Stat(saTokenPath); os.IsNotExist(err) {
```

Comment:
is it sufficient enough to check that the env variables are populated with this?

Should we have an equivalent of `inCluster` function here and check for env variable to be safe?

Reply:
> is it sufficient enough to check that the env variables are populated with this?

If the SA token has synced that means kube scheduled staticsync so yes.

>Should we have an equivalent of inCluster function here and check for env variable to be safe?

Where? we are waiting.

Reply:
to be clear static sync is a daemonset

Reply:
our container will not change its ENV unless it restarts right?

Reply:
yes ideally it will not change, but want to be sure with something like

```
if !inCluster {
    fmt.Errorf("kube env not populated")
}
```

Where inCluster is 
```
func inCluster() bool {
	if os.Getenv("KUBERNETES_SERVICE_HOST") == "" || os.Getenv("KUBERNETES_SERVICE_PORT") == "" {
		return false
	}
	return true
}
```

We can either do it after the wait loop, so the initContainer will crashlooping until the env is populated.

if the env is not populated, it is not a bug against us